### PR TITLE
feat: add bin `mock_testnet`

### PR DESCRIPTION
**Pull request by silathdiir**
### TODO

Tested prove type `block` with alpha URL `http://goerli-l2geth-rpc0.scroll.tech:8545`. Need to test with new deployed URL.

### Summary

1. Add a bin `mock_testnet` and it could be run via `make mock-testnet`.

2. Iterate to call l2geth API `l2_getTracesByBatchIndex` to prove block traces of each batch (has not been tested).

3. Or iterate to call rollupscan API to get `start_batch_number` and `end_batch_number`, then call l2geth API `scroll_getBlockTraceByNumberOrHash` to get block traces and prove (same as above batch type).

4. Add below ENVs.
```
# Used for `mock-testnet`
PROVE_TYPE=block # batch or block, block as default.
PROVE_BEGIN_BATCH=
PROVE_END_BATCH=
SCROLL_API_URL=
ROLLUPSCAN_API_URL=
```

### CODE REVIEW

File: bin/src/mock_prove.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+use types::eth::BlockTrace;
+use zkevm::circuit::SuperCircuit;
+use zkevm::prover::Prover;
+
+const DEFAULT_BEGIN_BATCH: i64 = 1;
+const DEFAULT_END_BATCH: i64 = i64::MAX;
+
+#[tokio::main]
+async fn main() {
+    log::info!("mock_prove: begin");
```
Comment (lispc) [inline bin/src/mock_prove.rs:12]: the log::info before env_logger::init() will not print out?  Either println or put it after  env_logger::init();

Comment (silathdiir) [reply bin/src/mock_prove.rs]: Moved this log after initializing `dotenv` and `env_logger` in https://github.com/scroll-tech/scroll-zkevm/pull/99/commits/dfc9b749e3a795d2eee24f4f321dcbfbff702c30.

File: Makefile [hunk 2]
```diff
@@ -40,6 +40,9 @@ mock:
 mock_pack:
 	@cargo test --features prove_verify --release test_mock_prove_all_target_circuits_packing -- --exact --nocapture
 
+mock_prove:
```
[the remark this answers is not in the document]
Comment (silathdiir) [reply Makefile]: Renamed `mock_prove` to `mock_testnet` in https://github.com/scroll-tech/scroll-zkevm/pull/99/commits/dfc9b749e3a795d2eee24f4f321dcbfbff702c30.

File: bin/src/mock_testnet.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+struct Setting {
+    begin_batch: i64,
+    end_batch: i64,
+    scroll_api_url: String,
+}
+
+impl Setting {
+    pub fn new() -> Self {
+        let scroll_api_url =
+            env::var("SCROLL_API_URL").expect("mock_prove: Must set env SCROLL_API_URL");
```
Comment (Lawliet-Chan) [inline bin/src/mock_testnet.rs:49]: I suggest we can `unwrap_or("http://localhost:8545")` here

**Comment by lispc** [comment]
@silathdiir since the "getBatchByNumber" api is still not ready, we can improve this PR so it can also mock prove 
 block by block,besides batch by batch.  It can support both mode.  You can ask xinran for the getBlockByNumber api format.

need to rebase against https://github.com/scroll-tech/scroll-zkevm/pull/118

@noel2004 is setting up a l2geth.

**Pull request merged**